lambda/main.tf

```diff
@@ -39,25 +39,22 @@ variable envvars {
   default = {}
 }
 
-resource "aws_iam_role" "role" {
-  name               = var.name
-  assume_role_policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Statement": [
-    {
-      "Action": "sts:AssumeRole",
-      "Principal": {
-        "Service": [
-            "lambda.amazonaws.com"
-        ]
-      },
-      "Effect": "Allow",
-      "Sid": ""
+data "aws_iam_policy_document" "role" {
+  version = "2012-10-17"
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+    principals {
```

Comment:
Did you test it? in JSON itself we have `Principal` (in singular) and here we have `principals` (in plural)

Reply:
(the same comment applies to `Action` vs `actions`)

Reply:
`principals` and `actions` are terraform keywords, that's how it works. They will be converted to appropriate `Action` and `Principal` JSON keys.